File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";

        int mini = Integer.MAX_VALUE;
        for (String s : strs) {
            mini = Math.min(mini, s.length());
        }

        StringBuilder result = new StringBuilder();

        for (int i = 0; i < mini; i++) {
            char ch = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if (strs[j].charAt(i) != ch) {
                    return result.toString();
                }
            }
            result.append(ch);
        }

        return result.toString();
    }
}

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";

        int mini = INT_MAX;
        for (auto it : strs) {
            mini = min(mini, (int)it.size());
        }

        string s = "";

        for (int i = 0; i < mini; i++) {
            char ch = strs[0][i];  
            for (int j = 1; j < strs.size(); j++) {
                if (strs[j][i] != ch) {
                    return s; 
                }
            }
            s += ch;  
        }

        return s;
    }
};
